### PR TITLE
Support encrypted data bags

### PR DESCRIPTION
**Pull request by michaelglass**
continuing mb-dev's work

### CODE REVIEW

**Comment by michaelglass** [comment]
seems to work.  only tested against v0 data bags and not node data bags.

**Comment by mb-dev** [comment]
Thank you for your efforts!

**Comment by michaelglass** [comment]
thoughts, @tobami?

File: libraries/search/overrides.rb [hunk 1]
```diff
@@ -86,10 +91,33 @@ def search_roles(_query, start, rows, &block)
       return _result
     end
 
+    def load_data_bag(bag_name, bag_item_id)
+      if Chef::Config[:encrypted_data_bag_secret]
+        begin
+          bag_item = Chef::EncryptedDataBagItem.load(bag_name, bag_item_id).to_hash
+        rescue Chef::EncryptedDataBagItem::DecryptionFailure,Chef::EncryptedDataBagItem::UnacceptableEncryptedDataBagItemFormat
```
Comment (michaelglass) [inline libraries/search/overrides.rb:98]: I tried to constrain this as tightly possible.

**Comment by michaelglass** [comment]
bump?

**Comment by thekorn** [comment]
Hi @michaelglass, hi @mb-dev,
thanks for your work on this feature, and sorry for not responding in a timely manner ...

I just tried to run the tests for this branch, and I got alot of failures, see below.
the question is, are the requirements mentioned in the README still valid?

```
* ruby >= 1.8
* ruby-chef >= 0.10.4
```

(I'm using exactly the lowest requirements)

Which are the versions of ruby and/or chef which make this tests pass for you?

Markus

**Comment by michaelglass** [comment]
The oldest release of ruby that's in the `.travis.yml` is `1.8.7`.  The oldest version of chef is `10.26.0`

**Comment by michaelglass** [comment]
to be clear: travis runs tests against ruby 1.8.7, 1.9.2, and 1.9.3, testing against chef 10.28.0 and 11.6.0 (see: https://travis-ci.org/edelight/chef-solo-search/)

**Comment by michaelglass** [comment]
bump bump?

**Comment by michaelglass** [comment]
also: curious if master passes with ruby 1.8.6

**Comment by a-chernykh** [comment]
Can't wait to see this merged. Thanks for an awesome work @mb-dev and @michaelglass I am using your branch in my `Berksfile` for now and it works like magic:

``` ruby
# https://github.com/edelight/chef-solo-search/pull/38
cookbook 'chef-solo-search', git: 'git@github.com:michaelglass/chef-solo-search.git', branch: 'support_encrypted_data_bags'
```

**Comment by tobami** [comment]
@michaelglass you really mean 1.8._6_? or would 1.8.7 work for you?

In any case, travis builds the branch on 1.8.7: https://travis-ci.org/edelight/chef-solo-search/builds/12259421 (41.6)

**Comment by michaelglass** [comment]
I'm going to merge this and update the readme to state that only travis tested builds are supported.

**Comment by a-chernykh** [comment]
Really glad to see this merged. Thanks for awesome work! Any plans to release new version? Looks like previous one was released 6 months ago.

**Comment by michaelglass** [comment]
good call @andreychernih

**Pull request merged**